### ISSUE TITLE
try optimize resnet50 instead of latent code

### ISSUE THREAD

**Issue by xiaohaipeng** [pbaylies/stylegan-encoder#36]
after fintuning resnet，load resnet and generator together，then optimize resnet，can this way have better results？

**Comment by pbaylies** [comment]
I'm not quite sure what you're proposing, but I'm welcome to proofs of concept or code! The resnet is mainly used to provide an estimate of the latents as a starting point for encoding.

**Comment by xiaohaipeng** [comment]
i mean after you have an estimate of the latents,connect the resnet and genarator , finetune resnet to get the best latents.
and i find after optimize the latent,generated-image looks blurry compared to the image generated by stylegan,does loss function lead to the effect?

**Comment by pbaylies** [comment]
You could try that, but you'd just end up with a resnet model that's fine-tuned for a single image.

**Comment by xiaohaipeng** [comment]
@pbaylies line 125 in perceptual_model.py,why sess.run([self.features_weight.initializer, self.features_weight.initializer]) twice?

**Comment by pbaylies** [comment]
For better or for worse, this was in @Puzer's original code as well.

**Comment by xiaohaipeng** [comment]
@pbaylies and i also want you help me make sure some details about stylegan:for images of1024 size,the w space is of[batchsize,18,512],after truncation ,get w+space of the same shape,how different of the two spaces?

**Comment by pbaylies** [comment]
Those are both w+, all truncation does is pull it more towards the center of the distribution.

**Comment by xiaohaipeng** [comment]
and，why do you use L1 loss on vgg features but use L2 loss on image pixels?what is your considerations when you design the five losses?

**Comment by pbaylies** [comment]
Trial and error :)

**Comment by xiaohaipeng** [comment]
haha,you must try much losses.and images  generated by stylegan always have defects or flaws,what reason do you think cause that?
